### PR TITLE
[2.0] Add the configuration options documentation for v2

### PR DESCRIPTION
**Pull request by ste93cry**
This PR adds the documentation about the configuration options

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    // Before Sentry 1.9 the private DSN must be used
+    $configuration = new Configuration(['server' => 'http://public:private@example.com/1']);
+
+    // After Sentry 1.9 the public DSN must be used
+    $configuration = new Configuration(['server' => 'http://public@example.com/1']);
+
+By default this option is borrowed from the ``SENTRY_DSN`` environment variable,
+if it exists. A value of ``null`` disables completely the sending of any event.
+Since Sentry 1.9 the public DSN must be used instead of the private one, which
+is deprecated and whose support will be removed in future releases of the server.
```
Comment (stayallive) [inline docs/configuration.rst]: I really think there should be at least an alias for the `dsn` key, although I think `server` should be the alias (or not used at all).

DSN is the common language to refer to this information for every integration AFAIK.

File: docs/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+By default this option is set to 6.
+
+Server
+======
+
+The DSN of the Sentry server the authenticated user is bound to.
+
+.. code-block:: php
+
+    // Before Sentry 1.9 the private DSN must be used
```
Comment (stayallive) [inline docs/configuration.rst]: Sentry 9 is the version where this is deprectated 😉 (1.9 mentioned a few more times)

File: docs/configuration.rst [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+is deprecated and whose support will be removed in future releases of the server.
+
+Server name
+===========
+
+The name of the server the SDK is running on (it's usually the hostname).
+
+.. code-block:: php
+
+    $configuration = new Configuration(['server_name' => 'foo']);
```
Comment (Jean85) [inline docs/configuration.rst]: Can we use a meaningful value in the example? Like `www.my-site.com`

File: docs/configuration.rst [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+Project root
+============
+
+The root of the project source code. As Sentry is able to distinguish project
+files from third-parties ones (e.g. vendors), this option can be configured to
+mark the directory containing all the source code of the application.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['project_root' => '/foo/bar']);
```
Comment (Jean85) [inline docs/configuration.rst]: Can we use ``project-folder/src/` in this example too?

File: docs/configuration.rst [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+Current environment
+===================
+
+The name of the current environment. There can be multiple environments per
+application, and each event belongs to one of them.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['current_environment' => 'foo']);
```
Comment (Jean85) [inline docs/configuration.rst]: Let's use a meaningful example here too, like `staging`

File: docs/configuration.rst [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    $configuration->setCurrentEnvironment('foo');
+
+Environments
+============
+
+The environments are a feature that allows events to be easily filtered in
+Sentry. An application can have multiple environments, but just one is active
+at the same time. This option let you configure the environments names: if
```
Comment (Jean85) [inline docs/configuration.rst]: What's the default behavior here? What's the default value?

File: docs/configuration.rst [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+
+    $configuration = new Configuration(['encoding' => 'json']);
+    $configuration->setEncoding('json');
+
+By default this option is set to ``gzip``.
+
+Context lines
+=============
+
+This option sets the number of lines of code context to capture. If ``null`` is
```
Comment (Jean85) [inline docs/configuration.rst]: What's the default value here?

File: docs/configuration.rst [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+By default this option is set to ``true``.
+
+Excluded exceptions
+===================
+
+Sometimes you may want to skip capturing certain exceptions. This option sets
+the FCQN of the classes of the exceptions that you don't want to capture. The
+check is done using the ``instanceof`` operator against each item of the array
+and if at least one of them passes the event will be recorded.
```
Comment (Jean85) [inline docs/configuration.rst]: Isn't the opposite?

File: docs/configuration.rst [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+Prefixes
+========
+
+This option sets the list of prefixes which should be stripped from the filenames
+to create relative paths.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['prefixes' => ['foo']);
```
Comment (Jean85) [inline docs/configuration.rst]: Let's use a meaningful example here too, like `/var/www`

File: docs/configuration.rst (hunk 8 above)
Comment (ste93cry) [reply docs/configuration.rst]: You're right, fixed

File: docs/configuration.rst (hunk 6 above)
Comment (ste93cry) [reply docs/configuration.rst]: I've added a description of the default behavior and of its default value

File: docs/configuration.rst (hunk 7 above)
Comment (ste93cry) [reply docs/configuration.rst]: I've added the default value

**Comment by ste93cry** [comment]
Travis is failing due to unrelated changes. I've addressed all the comments

File: docs/configuration.rst (hunk 1 above)
Comment (ste93cry) [reply docs/configuration.rst]: I strongly agree with you, however the change is out of scope of this PR which is only for documentation. I will open a new one to address the comment asap

**Comment by ste93cry** [comment]
@stayallive Can we get your approval on the PR if everything is alright for you?

**Comment by stayallive** [comment]
@ste93cry, yes sorry 👍 missed the changed made 💯

**Pull request merged**